### ISSUE TITLE
VersionReq to_string does not retain patch in output

### ISSUE THREAD

**Issue by frewsxcv** [dtolnay/semver#116]
Found this [while fuzzing this crate](https://github.com/rust-fuzz/targets/pull/75).

```rust
extern crate semver;

fn main() {
    let str_input = "8.*.7";
    println!("input:\n\t{:?}", str_input);

    let parsed1 = semver::VersionReq::parse(str_input).unwrap();
    println!("parsed:\n\t{:?}", parsed1);

    let version_req_s = parsed1.to_string();
    println!("parsed to_string:\n\t{:?}", version_req_s);

    let parsed2 = semver::VersionReq::parse(&version_req_s).unwrap();
    println!("parsed to_string parsed:\n\t{:?}", parsed2);
}
```

```
input:
	"8.*.7"
parsed:
	VersionReq { predicates: [Predicate { op: Wildcard(Minor), major: 8, minor: None, patch: Some(7), pre: [] }] }
parsed to_string:
	"8.*"
parsed to_string parsed:
	VersionReq { predicates: [Predicate { op: Wildcard(Minor), major: 8, minor: None, patch: None, pre: [] }] }
```

**Comment by steveklabnik** [comment]
Soooooo I thought that this was just malformed in general....

**Comment by dtolnay** [comment]
Relevant issue for rejecting `8.*.7` upstream in semver-parser: https://github.com/steveklabnik/semver-parser/issues/62

**Comment by dtolnay** [comment]
Fixed in 1.0.0.

```rust
fn main() {
    println!("{}", semver::VersionReq::parse("8.*.7").unwrap_err());
}
```

```console
unexpected character after wildcard in version req
```
